low-level/files/opt/seagate/sspl/bin/sspl_setup

```diff
@@ -1,97 +1,234 @@
+        self._script_dir = os.path.dirname(os.path.abspath(__file__))
+
+    @property
+    def args(self) -> str:
+        return self._args
+
+    @staticmethod
+    def usage(prog: str):
+        """Print usage instructions"""
+
+        sys.stderr.write(
+            f"{prog} [post_install[-p <LDR_R1>]|init [-dp] [-r <ssu|gw|cmu|vm>]\n|config [-f] [-r <ssu|gw|cmu|vm>]|test [self|sanity]|reset [hard -p <LDR_R1>|soft]]\n"
+            "init options:\n"
+            "\t -dp Create configured datapath\n"
+            "\t -r  Role to be configured on the current node\n"
+            "config options:\n"
+            "\t -f  Force reinitialization. Do not prompt\n"
+            "\t -r  Role to be configured on the current node"
+            "post_install options:\n"
+            "\t -p Product to be configured\n"
+            "reset options:\n"
+            "\t -p Product to be configured\n")
+
+    @staticmethod
+    def get_command(desc: str, argv: dict):
+        """Return the Command after parsing the command line."""
+
+        parser = argparse.ArgumentParser(desc)
+        subparsers = parser.add_subparsers()
+        cmds = inspect.getmembers(sys.modules[__name__])
+        cmds = [(x, y) for x, y in cmds
+            if x.endswith("Cmd") and x != "Cmd"]
+        for name, cmd in cmds:
+            cmd.add_args(subparsers, cmd, name)
+
+        args, unknown = parser.parse_known_args(argv)
+        args.args = unknown + args.args
+        return args.command(args)
+
+    @staticmethod
+    def add_args(parser: str, cls: str, name: str):
+        """Add Command args for parsing."""
+
+        parsers = parser.add_parser(cls.name, help='setup %s' % name)
+        parsers.add_argument('args', nargs='*', default=[], help='args')
+        parsers.set_defaults(command=cls)
+
+    @staticmethod
+    def _send_command(command: str, fail_on_error=True):
+        process = subprocess.Popen(
+            command, shell=True, stdout=subprocess.PIPE, stderr=subprocess.PIPE)
+        response, error = process.communicate()
+        if error is not None and \
+        len(error) > 0:
+            print("command '%s' failed with error\n%s" % (command, error))
+            if fail_on_error:
+                sys.exit(1)
+            else:
+                return str(error)
+        return str(response)
+
+    # @staticmethod
+    # def _call_script(script_dir: str, name: str, args: list):
+    #     script_path = f"{script_dir} {name} {' '.join(args)}"
+    #     Cmd._send_command(script_path)
+
+
+class PostInstallCmd(Cmd):
+    """PostInstall Setup Cmd"""
+    name = "post_install"
+    script = "sspl_post_install"
+
+    def __init__(self, args: dict):
+        super().__init__(args)
+
+    def process(self):
+        # TODO: Import relevant python script here for further execution.
+        print(f"{self._script_dir}/{self.script}", self.name, ' '.join(self._args))
+        # Cmd._call_script(f"{self._script_dir}/{self.script}", self.name, self._args)
```

Comment:
Add pass
And remove _call_script() from final commit

Reply:
Done